pkg/kubelet/eviction/api/types.go

```diff
@@ -32,10 +32,17 @@ const (
 	SignalNodeFsAvailable Signal = "nodefs.available"
 	// SignalNodeFsInodesFree is amount of inodes available on filesystem that kubelet uses for volumes, daemon logs, etc.
 	SignalNodeFsInodesFree Signal = "nodefs.inodesFree"
-	// SignalImageFsAvailable is amount of storage available on filesystem that container runtime uses for storing images and container writable layers.
+	// ImageFs and ContainerFs can differ what is stored based on container runtime configuration
+	// In cases of ContainerFs being on root, then nodefs = containerfs
+	// ImageFs would then be the separate disk for image layers.
+	// SignalImageFsAvailable is amount of storage available on filesystem that container runtime uses for storing images layers.
 	SignalImageFsAvailable Signal = "imagefs.available"
-	// SignalImageFsInodesFree is amount of inodes available on filesystem that container runtime uses for storing images and container writable layers.
+	// SignalImageFsInodesFree is amount of inodes available on filesystem that container runtime uses for storing images layers.
 	SignalImageFsInodesFree Signal = "imagefs.inodesFree"
+	// SignalContainerFsAvailable is amount of storage available on filesystem that container runtime uses for container writable layers.
+	SignalContainerFsAvailable Signal = "containerfs.available"
```

Comment:
We probably want to disallow these being specified and make them the same as the nodefs values.

Reply:
Well its either imagefs or nodefs values. I'll add a way to specify this.

Reply:
This was a lot more complicated than just making them the same as nodefs.  I ignore user configurations for containerfs.available and set them to either nodefs or imagefs values.  

I added some unit tests in helpers.test for each and every case.

I add a warning and override them.  We don't really validate the evictionHard/evitionSoft settings in the api validation so I'm not sure if we can disallow.  I mainly override based on container runtime configuration.